infrastructure/code/snyk_code_http_client.go

```diff
@@ -434,11 +446,11 @@ func (s *SnykCodeHTTPClient) analysisRequestBody(options *AnalysisOptions) ([]by
 	return requestBody, err
 }
 
-func (s *SnykCodeHTTPClient) checkResponseCode(r *http.Response) error {
-	if r.StatusCode >= 200 && r.StatusCode <= 299 {
+func (s *SnykCodeHTTPClient) checkResponseCode(statusCode int) error {
+	if statusCode >= 200 && statusCode <= 400 {
```

Comment:
This has consequences for redirects. If we're not currently equipped to follow them, maybe we should treat them as "unexpected"?

Reply:
What do you mean with `unexpected`? Why can't we follow the redirects? Doesn't the http client follow them unless we deactivate it? I definitely see an error though that it's `400` and not `399`.

Reply:
I forgot net/http.Client automatically follows them (if you use the default Client settings). Nevermind.

Good catch on the 400, I'd completely missed that. What caused me to comment on this originally, an drive-by change like this can introduce other side-effects, which are offtopic from the original PR (add logging). So maybe this is best left for a separate PR?

Reply:
It was in the spirit of handling the HTTP 400, but I agree, this is something we should separate out.